### PR TITLE
gtk-sharp3: fixed build and removed broken flag.

### PR DESCRIPTION
**Pull request by dasj19**
###### Motivation for this change
Removed broken flag from gtk-sharp-2.99.3 and fixed the compiling issues that made it broken.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
CC: @prusnak

### CODE REVIEW

File: pkgs/development/libraries/gtk-sharp/3.0.nix [hunk 1]
```diff
@@ -13,30 +13,29 @@
...
 , monoDLLFixer
 }:
-
-stdenv.mkDerivation {
-  name = "gtk-sharp-2.99.3";
+let
+  pname = "gtk-sharp";
+  version = "2.99.3";
+in stdenv.mkDerivation {
+  name = "${pname}-${version}";
```
Comment (prusnak) [inline pkgs/development/libraries/gtk-sharp/3.0.nix:20]:
```suggestion
stdenv.mkDerivation rec {
  pname = "gtk-sharp";
  version = "2.99.3";
```

File: pkgs/development/libraries/gtk-sharp/3.0.nix [hunk 2]
```diff
@@ -13,30 +13,29 @@
...
-  # '';
-
   nativeBuildInputs = [ pkgconfig ];
   buildInputs = [
     mono glib pango gtk3 GConf libglade libgnomecanvas
     libgtkhtml libgnomeui libgnomeprint libgnomeprintui gtkhtml libxml2
   ];
 
+  patchPhase = ''
+    substituteInPlace gtk/gui-thread-check/profiler/gui-thread-check.c \
```
Comment (jtojnar) [inline pkgs/development/libraries/gtk-sharp/3.0.nix:35]: Maybe better use https://github.com/mono/gtk-sharp/pull/266

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/gtk-sharp/3.0.nix (hunk 2 above)
Comment (dasj19) [reply pkgs/development/libraries/gtk-sharp/3.0.nix]: @jtojnar Added that patch to the repo.

File: pkgs/development/libraries/gtk-sharp/3.0.nix (hunk 1 above)
Comment (dasj19) [reply pkgs/development/libraries/gtk-sharp/3.0.nix]: I made the requested change. Thanks

**Review by prusnak — APPROVED** [review]

File: pkgs/development/libraries/gtk-sharp/3.0.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/development/libraries/gtk-sharp/3.0.nix]: Is it not possible to fetch it with `fetchpatch`?

Comment (dasj19) [reply pkgs/development/libraries/gtk-sharp/3.0.nix]: I did not know about fetchpatch, and I don't know how to get a patch file from this: https://github.com/mono/gtk-sharp/pull/266/files

Comment (jtojnar) [reply pkgs/development/libraries/gtk-sharp/3.0.nix]: You can just add `.patch` after the URL, for example here you would fetch `https://github.com/mono/gtk-sharp/commit/c1b9bd4214b78dc426bc6e6a07c23a8ab4ab50ea.patch`

Comment (dasj19) [reply pkgs/development/libraries/gtk-sharp/3.0.nix]: wow, nice trick... I'll do a fetchpatch then.

Comment (dasj19) [reply pkgs/development/libraries/gtk-sharp/3.0.nix]: I noticed that when I `nix-prefetch-url https://github.com/mono/gtk-sharp/commit/401df51bc461de93c1a78b6a7a0d5adc63cf186c.patch` I get a different hash than the nix expression evaluator ... do you know why is that happening?

Comment (jtojnar) [reply pkgs/development/libraries/gtk-sharp/3.0.nix]: `fetchpatch` also cleans the patch up so you would need a hash of the cleaned file.

File: pkgs/development/libraries/gtk-sharp/3.0.nix [hunk 3]
```diff
@@ -14,29 +18,31 @@
 , monoDLLFixer
 }:
 
-stdenv.mkDerivation {
-  name = "gtk-sharp-2.99.3";
+stdenv.mkDerivation rec {
+  pname = "gtk-sharp";
+  version = "2.99.3";
+  name = "${pname}-${version}";
```
Comment (jtojnar) [inline pkgs/development/libraries/gtk-sharp/3.0.nix:24]: This is redundant, `mkDerivation` does this by default.

File: pkgs/development/libraries/gtk-sharp/3.0.nix [hunk 4]
```diff
@@ -14,29 +18,31 @@
...
-  # '';
-
   nativeBuildInputs = [ pkgconfig ];
   buildInputs = [
     mono glib pango gtk3 GConf libglade libgnomecanvas
     libgtkhtml libgnomeui libgnomeprint libgnomeprintui gtkhtml libxml2
   ];
 
+  patches = [
+    (fetchpatch {
```
Comment (jtojnar) [inline pkgs/development/libraries/gtk-sharp/3.0.nix:39]: Could you add a comment with explanation and link to the PR that produced the commit?

Comment (dasj19) [reply pkgs/development/libraries/gtk-sharp/3.0.nix:40]: Done. Thanks.

**Review by prusnak — APPROVED** [review]

**Comment by jtojnar** [comment]
Thank you.

**Pull request merged**